metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -132,6 +133,9 @@ public void execute(Tuple tuple) {
         Optional<List<JSONObject>> messages = parser.parseOptional(originalMessage);
         for (JSONObject message : messages.orElse(Collections.emptyList())) {
           message.put(Constants.SENSOR_TYPE, getSensorType());
+          if(!message.containsKey(Constants.GUID)) {
```

Comment:
You should probably do this after the field transformations, so that if people provide their own GUID, they don't have one imposed on them.